zerver/data_import/slack.py

```diff
@@ -898,6 +899,38 @@ def get_messages_iterator(
         yield from sorted(messages_for_one_day, key=get_timestamp_from_message)
 
 
+def get_parent_user_id_from_thread_message(thread_message: ZerverFieldsT, subtype: str) -> str:
+    """
+    This retrieves the user id of the sender of the original thread
+    message.
+    """
+    if subtype == "thread_broadcast":
+        return thread_message["root"]["user"]
+    elif thread_message["thread_ts"] == thread_message["ts"]:
+        # This is the original thread message
+        return thread_message["user"]
+    else:
+        return thread_message["parent_user_id"]
+
+
+def get_zulip_thread_topic_name(
+    message: ZerverFieldsT, thread_ts: datetime, thread_counter: dict[str, int]
+) -> str:
+    """
+    The topic name format is date + message snippet + counter.
+
+    e.g "2024-05-22 Hello this is a long message that will be c... (1)"
+    """
+    thread_date = thread_ts.strftime(r"%Y-%m-%d")
+    THREAD_TOPIC_SNIPPET_LENGTH = 46 + len(thread_date)
+    base_zulip_topic_name = f"{thread_date} {message['text']}"
+    collision = thread_counter[base_zulip_topic_name]
+    thread_counter[base_zulip_topic_name] += 1
+    count = (f" ({collision + 1})") if collision > 0 else ""
+    topic_name = truncate_content(base_zulip_topic_name, THREAD_TOPIC_SNIPPET_LENGTH, "...")
```

Comment:
You can use a unicode `...`  character to save a couple characters of real content when truncating; they also just look nicer.